packages/package-json-lint/src/rules/require-no-hard-coded-dependency-versions.js

```diff
@@ -0,0 +1,23 @@
+module.exports = {
+  create: ({ ruleConfig, projectType, report }) => ({
+    dependencies: (dependencies) => {
+      if (projectType === 'module' || projectType === 'devModule') {
+        const messageString = 'no hard coded dependency';
+        const currentProblems = Object.keys(dependencies).map(dependencyName => {
+          const dependencyVersion = dependencies[dependencyName];
+          if (!dependencyVersion.startsWith('^') && !(ruleConfig.severity.allowList && ruleConfig.severity.allowList.includes(dependencyName))) {
+            return `${dependencyName}@${dependencyVersion} does not satisfy requirement for ${messageString}`;
+          }
+          return undefined;
+        }).filter(problem => !!problem);
+
+        if (currentProblems.length) {
+          const lintMessage = `The dependencies for this project have hard-coded versions for ${messageString}:\n  ${currentProblems.join('\n  ')}`;
```

Comment:
Similar to the above comment, this message sounds a bit confusing.
`The dependencies for this project have hard-coded versions for no hard coded dependency:`

Should we say something like:
`The dependencies for this project have hard-coded versions that violates the require-no-hard-coded-dependency-versions rule:`

Reply:
Updated here https://github.com/cerner/terra-toolkit/pull/704/commits/b3446c5a790a1172c0c86da818d7d21076835b53.